engine/src/main/java/com/datatorrent/stram/client/StramAgent.java

```diff
@@ -417,11 +417,25 @@ private StramWebServicesInfo retrieveWebServicesInfo(String appId)
       String appPath = response.getString("appPath");
       String user = response.getString("user");
       JSONObject permissionsInfo = null;
-      try (FSDataInputStream is = fileSystem.open(new Path(appPath, "permissions.json"))) {
-        permissionsInfo = new JSONObject(IOUtils.toString(is));
+      Path permissionsPath = new Path(appPath, "permissions.json");
+      LOG.debug("Checking for permission information in file {}", permissionsPath);
+      try {
+        if (fileSystem.exists(permissionsPath)) {
+          LOG.info("Loading permission information");
+          try (FSDataInputStream is = fileSystem.open(permissionsPath)) {
+            permissionsInfo = new JSONObject(IOUtils.toString(is));
+          }
+        } else {
+          // ignore and log messages if file is not found
+          LOG.info("Permission information not available, file not present");
+        }
       } catch (IOException ex) {
-        // ignore if file is not found
-        LOG.info("Exception in accessing permissions.json", ex);
+        // ignore and log message when unable to read the file
+        if (LOG.isDebugEnabled()) {
```

Comment:
The exception can only be raised if something is wrong with the file system or caller does not have access to the specified path. In both cases, it will be OK to log the exception.